internal/pkg/instrumentation/bpf/google.golang.org/grpc/server/probe.go

```diff
@@ -72,6 +83,14 @@ func New(logger *slog.Logger) probe.Probe {
 				Sym:        "google.golang.org/grpc/internal/transport.(*http2Server).operateHeaders",
 				EntryProbe: "uprobe_http2Server_operateHeader",
 			},
+			{
+				Sym:        "google.golang.org/grpc/internal/transport.(*serverHandlerTransport).WriteStatus",
+				EntryProbe: "uprobe_serverHandlerTransport_WriteStatus",
+			},
+			{
+				Sym:        "google.golang.org/grpc/internal/transport.(*http2Server).WriteStatus",
+				EntryProbe: "uprobe_http2Server_WriteStatus",
+			},
```

Comment:
Are we instrumenting both of them because either of them might be used?

Reply:
It needs to be a struct.
The point I think worth verifying is whether: a) one of these implementations will be called per stream b) both of them will be called and each is a different layer.
The first case sounds more reasonable but worth making sure with a debugger. WDYT?

Reply:
@RonFed I wasn't sure how I would trigger one or the other, so I don't think I could get that reliably from a debugger. But I did open https://github.com/grpc/grpc-go/issues/7730 to ask the grpc maintainers and got this reply that seems to answer it: https://github.com/grpc/grpc-go/issues/7730#issuecomment-2406553932

Sounds like http2server is the default gRPC implementation, and serverHandlerTransport is used for integrating gRPC into existing HTTP handlers. Given our goals (and the fact that we only probe http2server for `operateHeaders`), I think we can just go with that.

In the future, we might want to support serverHandlerTransport for the integration use case. But for OOTB gRPC support, http2 should be good if I understand. Will update the PR to just use that one

Reply:
Sounds good, thanks for checking